src/assets/scripts/aircraft/AircraftInstanceModel.js

```diff
@@ -1705,6 +1705,42 @@ export default class Aircraft {
     }
 
     /**
+     * Checks if the aircraft is inside the airspace of a specified airport
+     *
+     * @for AircraftInstanceModel
+     * @method isInsideAirspace
+     * @param  {airport} airport the airport whose airspace we are checking
+     * @return {Boolean}
+     * @private
+     */
+    isInsideAirspace(airport) {
+        let withinAirspaceLateralBoundaries = this.distance <= airport.ctr_radius;
+        const withinAirspaceAltitudeRange = this.altitude <= airport.ctr_ceiling;
+
+        if (!_isNil(airport.perimeter)) {    // polygonal airspace boundary
+            withinAirspaceLateralBoundaries = point_in_area(this.position, airport.perimeter);
+        }
+
+        return withinAirspaceAltitudeRange && withinAirspaceLateralBoundaries;
+    }
+
+    /**
+     * Aircraft has "weight-on-wheels" (on the ground)
+     * @for AircraftInstanceModel
+     * @method isOnGround
+     */
+    isOnGround() {
+        const error_allowance_ft = 5;
+        const apt = window.airportController.airport_get();
+        const rwy_elev = apt.getRunway(this.rwy_dep || this.rwy_arr).elevation;
```

Comment:
two things here: 
- `apt.getRunway(this.rwy_dep || this.rwy_arr).elevation` is confusing.  is this something `getRunway` could do internally?
- although not bad, I'm not a fan of the `someMethod().property` style. I know this is littered throughout the app and it works, I just find it confusing.  if you need a property from the return of a method destructure it or use the returned object: 
```javascript
// destructure
const { elevation } = airport.getRunway();

// use object
const runway = airport.getRunway();
runway.elevation
```
In most cases I would lean towards the second option just for readability.